### PR TITLE
[FLINK-31832] Add benchmarks for end to end restarting tasks.

### PR DESCRIPTION
**Pull request by huwh**

### CODE REVIEW

File: src/main/java/org/apache/flink/scheduler/benchmark/e2e/HandleGlobalFailureAndRestartAllTasksBenchmarkExecutor.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Setup(Level.Trial)
+    public void setup() throws Exception {
+        benchmark = new HandleGlobalFailureAndRestartAllTasksBenchmark();
+        benchmark.setup(jobConfiguration);
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.SingleShotTime)
+    public void startSchedulingAndFailover() throws Exception {
```
Comment (wanglijie95) [inline src/main/java/org/apache/flink/scheduler/benchmark/e2e/HandleGlobalFailureAndRestartAllTasksBenchmarkExecutor.java:54]: `startSchedulingAndFailover` -> `handleGlobalFailureAndRestartAllTasks`

**Review by wanglijie95 — APPROVED** [review]
Thanks for creating this PR @huwh. I have only one minor comment, PTAL

**Comment by huwh** [comment]
comment addressed. Thank you for your review.

**Review by wanglijie95 — APPROVED** [review]
Thanks @huwh, LGTM.

**Pull request merged**